apps/src/lab2/levelEditors/aichatSettings/EditAichatSettings.tsx

```diff
@@ -88,13 +89,30 @@ const EditAichatSettings: React.FunctionComponent<{
     [aichatSettings, setAichatSettings, initialCustomizations]
   );
 
+  const setModelSelectionValues = useCallback(
```

Comment:
Not a strong feeling, but what do you think about separating this function into two different functions - one to update the `selectedModelId` and the other to update the `availableModelIds`? I think it may make things a bit easier to read where these functions are called. Also, one property is included in `initialCustomizations` and the other property is not.

Reply:
I thought so too, and I was originally using setPropertyValue for the selectedModelId in initialCustomizations, but since these are still in the same piece of state, they weren't reliably updating correctly, so I had to combine them to set both values at the same time. My next commit will make the code a bit simpler tho at least.

Reply:
Oh interesting. I actually had a similar issue with updating back when working on edubot project. And thanks for iterating on making code more simple!